enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,132 @@
+
+### Non-Goals
+
+- Provide generic metrics about OLM managed operators
+- Own the process to integrate with Telemeter
+- Support operand metric reporting
+
+## Proposal
+
+### Generic Prometheus Operator Support
+
+#### OLM support for new Prometheus objects in bundles
+
+The bundle object currently has the following format:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC resources. This enhancement proposes adding support for Prometheus `ServiceMonitors` and `PrometheusRules` resources:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+OLM would then need to be updated to identify the `ServiceMonitors` and `PrometheusRules` resources and deploy them as part of an `InstallPlan`. There is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the appropriate RBAC privileges to view the `ServiceMonitor` and `PrometheusRule` resource events in the namespace that the operator is being deployed to. As such, the cluster admin will need to configure the Prometheus Operator to watch for events in the correct namespace.
+
+### OpenShift Monitoring Prometheus Operator Support
+
+#### Requirements
+
+Once OLM offers bundle support for the `ServiceMonitor` and `PrometheusRule` resources, a sanctioned set of OLM managed operators should be able to report metrics to the OpenShift Monitoring Prometheus Operator.
+
+A number of requirements must be met for an application's metrics to be identified as cluster monitoring workload before it is scraped by the OpenShift Monitoring Prometheus Operator.
+
+##### Namespace Requirements
+
+The application must be deployed in a namespace that:
+
+- Is prefixed with `openshift-`
+- Is labeled with `openshift.io/cluster-monitoring=true`
+
+##### ServiceMonitor Requirements
+
+A `ServiceMonitor` object must be created in the namespace mentioned above and point to the application's metrics endpoint.
+
+>Note: This requirement is already fulfilled with OLM bundle support for Prometheus resources.
+
+##### RBAC Requirements
+
+A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC privileges to discover the `ServiceMonitor` and `PrometheusRule` resources in the newly created namespace.
+
+#### Fulfilling Namespace and RBAC Requirements
+
+OLM itself will not be responsible for fulfilling the Namespace and RBAC requirements. Instead, OLM will define how operator authors can provide OpenShift Console with the information required to fulfill these requirements.
```

Comment:
i would not expect the console to deal w/ RBAC.  The operator itself is going to need to grant those permissions.

Reply:
@awgreene can we get a follow up to resolve this?